src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -104,7 +105,9 @@ def __init__(
                 config=config,
             )
         else:
-            self._alloc_from_dict(config_dict=config_dict)
+            self._alloc_from_dict(
+                config_dict=config_dict, source_file=config_dict_source_file
+            )
```

Comment:
This seems a bit confusing to me, this seems to suggest that we only have dictionaries that are read from a file. Is this the intended design?

Reply:
I am wondering if this isn't less useful. Isn't it better that the list of defines is given explicitly?

Reply:
about dictionaries only read from file - it was my impression that there's where we moving towards / want to get in the end, and I did not consider other alternatives for getting a dictionary (besides from tests of course). do you have a more accurate naming suggestion, Dan?

Re: Eivind - we can totally change the interface and yank the creation of predefined defines out of resconfig, thanks for the suggestion! would you like me to let it live in...

so i'm gonna 
- change the interface of enkfmain, and let it accept a `predefined_defines: dict` argument, and
- move the convenience method for creating our current default to ...

hmm. i usually don't go higher up in the stack than enkf main, i see ResConfig is initialized

- in `__main__.py`,
- in `cli/main.py`, and
- in `libres_facade.py`

any suggestions as to where the method should live so it is accessible by all three?

Reply:
What we have thought about is getting a `LibresFacade.from_dict` classmethod analogous to `LibresFacade.from_file` that hides the implementation detail of initializing `ResConfig` and `EnKFMain` mind that this opens up the pandoras box of moving everest away from initializing using `ResConfig(config=...)`.

Reply:
> about dictionaries only read from file - it was my impression that there's where we moving towards / want to get in the end, and I did not consider other alternatives for getting a dictionary (besides from tests of course). do you have a more accurate naming suggestion, Dan?

Even though we are aiming at reading dictionaries from files we are not there yet and currently, there is no instance where a dictionary is read from a file (for ResConfig), so I am not sure it is a good idea to try and implement some distant future functionality now.

But let's say we want to try to do that I am not sure passing a dict and a file_name where that dict is being read from is a winning combination in any kind of future-proofing.  You would be either passing the filepath or the dict, if the implementation requires both then the implementation can be improved :)

My suggestion would be to focus on fixing the tests and not adding/changing functionality so that we can test it.

Reply:
i guess we can't really future proof at this stage, really I was just trying to move towards the right direction. at the moment, you're of course right, we don't parse a dict from file - but at the same time we generate config dicts for tests, and we want to test those using the code, putting them somewhere. to me, it feels right to think about how things might look like, and nudge the code towards that direction, rather than just nudging the code towards somehow running the tests. but of course the latter strategy might be just right, and we might end up with what we want the code to look like in the end...

> You would be either passing the filepath or the dict

Morten very passionately disagreed, arguing for a dictionary that truly reflects only the contents of the user config file. that necessitates to have something "come along" with the dictionary, to carry meta info, or context, which we need. Of course, one can just as well put everything in the dictionary. I dunno. I think separating it and having the separation clear and out in the open is a good idea, but there's a difference of opinions here...

> focus on fixing the tests and not adding/changing functionality
hmmm. yeah I mean if we can, that would be the best. I'm gonna have to think about that one. of course, I think that's what I originally tried to do - fix the tests, by stuffing the DEFINE keyword with predefineds in the generated config dicts, and using a test internal keyword. and we all know how much good that did me ; )

Reply:
Dan, for an alternative approach to passing the config file, see [my comment below](https://github.com/equinor/ert/pull/4215#discussion_r1021816113), or directly check out [Eivinds idea here](https://github.com/equinor/ert/pull/4215#issuecomment-1311225883)

Reply:
@valentin-krasontovitsch I think going down the route of implementing `LibresFacade.from_dict` and moving everest is too soon right now. Also updating the `cli/gui/everest/semeio` to a new `__init__` is kind of unnecessarily messy as we would rather go straight to `LibresFacade.from_dict/LibresFacade.from_file`.